### DISCUSSION TITLE
[SOLVED] Enabling grub.cfg automatic updates by monitoring snapshots path

### DISCUSSION THREAD

**Discussion by FGD-Garuda** [Antynea/grub-btrfs#130] [Q&A] [answered]
Hi there,

I wanted to enable the update of grub menus without performing `update-grub` manually every time there is a change in my snapshots path.

I did `sudo systemctl enable grub-btrfs.path` then `sudo systemctl start grub-btrfs.path` and the output of `sudo systemctl cat grub-btrfs.path` is

```
# /lib/systemd/system/grub-btrfs.path
[Unit]
Description=Monitors for new snapshots

[Path]
PathModified=/.snapshots

[Install]
WantedBy=multi-user.target
```

Am I missing something?
Cuz grub's menu does not update without me doing sudo update-grub manually.
I do my snapshots manually into subfolders of `/.snapshots`

Thanks

**Comment by Antynea** [comment, marked as answer]
Hello,

Nothing to do here.
I think you use kubuntu distribution
Edit `/etc/default/grub-btrfs/config`
Line 92:
`#GRUB_BTRFS_MKCONFIG=/usr/bin/grub2-mkconfig`
to
`GRUB_BTRFS_MKCONFIG=/usr/sbin/grub-mkconfig`

`grub-update` on a debian like distribution is an alias for grub-mkconfig
if `grub-mkconfig` isn't locate in `/usr/sbin/` use `find / -name grub-mkconfig` command to find it.

**Reply by FGD-Garuda** [reply]
True, I use Kubuntu, it is effectively located at `/usr/sbin/grub-mkconfig`

I edited the confg file, disabled, re-enabled, started, made changes in my snapshots. I see no change to .cfg files in /boot/grub, when is the service actually applying the changes to the grub menu, is it at every log-off/restart/shutdown or within the second a change is applied in the /.snapshots path?

**Comment by Antynea** [comment]
Within the second a change is applied in the /.snapshots path.

**Reply by FGD-Garuda** [reply]
I must be doing something wrong, I'll reboot and see if anything gets on the right track.

**Reply by Antynea** [reply]
You can use `systemctl status grub-btrfs.service` to see if an error appear when you create or delete a snapshot in `/.snapshots`

**Reply by FGD-Garuda** [reply]
Ha it's working!

Just before rebooting (of course **after** closing all my terminal windows and applications, darn it loll) I did another snapshots and now I see the automated update on `/boot/grub/grub-btrfs.cfg`. :)

It seems it needed 5-10mins after re-enabling the service to actually starting monitoring /.snapshots.

It's so great now, I love using my snapshots and now all is automated and my data is secured by my snapshots schedule and scripts and `grub-btrfs` itself. :) Of course I do backups on a second ssd, if one fails the other works in a jiffy.

Thanks for this man, so helpful! I hope a bunch of other users enjoy it too! BTRFS is great.

**Reply by Antynea** [reply]
Glad to hear that, have fun :)
